### PR TITLE
Refactor usageupdate

### PR DESCRIPTION
**Pull request by kjgarza**

### CODE REVIEW

**Comment by mfenner** [comment]
@kjgarza can you fix Travis errors?

File: app/models/usage_update.rb [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    msgs_available = req.attributes['ApproximateNumberOfMessages']
+    msgs_available.to_i
+  end
+
+  def queue_jobs(options={})
+
+    total = get_total(options)
+    
+    if total < 1
+      text = "No works found for in the Usage Reports Queue."
```
Comment (mfenner) [inline app/models/usage_update.rb]: Please fix language.

File: app/models/usage_update.rb [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    num_messages = total
+    while num_messages > 0 
+        queued = process_data(options)
+        num_messages -= queued
+        puts num_messages
+        puts queued
+    end
+    text = "#{queued} reports queued out of #{total} for Usage Reports Queue"
+
+    Rails.logger.info text
```
Comment (mfenner) [inline app/models/usage_update.rb]: Rails.logger only works in controllers. Please define a logger that logs to stdout.

File: app/models/usage_update.rb [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    return report.body.fetch("errors") if report.body.fetch("errors", nil).present?
+    return [{ "errors" => { "title" => "The report is blank" }}] if report.body.blank?
+
+    items = report.body.dig("data","report","report-datasets")
+    header = report.body.dig("data","report","report-header")
+    report_id = report.url
+
+    created = header.fetch("created")
+    Array.wrap(items).reduce([]) do |x, item|
+      data = {}
```
Comment (mfenner) [inline app/models/usage_update.rb]: Please rewrite to `data = { doi: item.dig("dataset-id").first.dig("value"), pid: normalize_doi(data[:doi]) }`, etc.

File: app/models/usage_update.rb [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+  def self.format_event type, data, options={}
+    fail "Not type given. Report #{data[:report_id]} not proccessed" if type.blank?
+    fail "Access token missing." if ENV['SASHIMI_SOURCE_TOKEN'].blank?
+    fail "Report_id is missing" if data[:report_id].blank?
+    { "uuid" => SecureRandom.uuid,
+      "message-action" => "create",
+      "subj-id" => data[:report_id],
+      "subj"=> {
+        "pid"=> data[:report_id],
```
Comment (mfenner) [inline app/models/usage_update.rb]: please use `id` instead of `pid`.

File: spec/factories/default.rb [hunk 5]
```diff
@@ -13,4 +13,21 @@
...
   end
+
+  factory :event , class: OpenStruct do
+    sequence(:uuid) { |n| "#{SecureRandom.uuid}-#{n}" }
+    message_action "create"
+    sequence(:obj_id) { |n| "#{Faker::Internet.url}#{n}" }
+    sequence(:subj_id) { |n| "#{Faker::Internet.url}#{n}" }
+    total Faker::Number.number(3)
+    subj {{
+      "pid" => "fdfdfd",
```
Comment (mfenner) [inline spec/factories/default.rb]: please use `id` instead of `pid`.

File: app/models/usage_update.rb [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+    "Queued import for Report updated from #{from_date.strftime("%F")} until #{until_date.strftime("%F")}."
+  end
+
+  def self.import(options={})
+    # from_date = options[:from_date].present? ? Date.parse(options[:from_date]) : Date.current - 1.day
+    # until_date = options[:until_date].present? ? Date.parse(options[:until_date]) : Date.current
+
+    usage_update = UsageUpdate.new
+    # usage_update.queue_jobs(usage_update.unfreeze(from_date: from_date.strftime("%F"), until_date: until_date.strftime("%F")))
+    usage_update.queue_jobs
```
Comment (mfenner) [inline app/models/usage_update.rb]: Does this work? No date range given, what is `import_by_month` doing?

Comment (kjgarza) [reply app/models/usage_update.rb]: At the moment Im not using `import_by_month` for anything. I took the crossref file as a template, that's why is there. 

You need to execute the rake task. And probably this is related to my question from earlier. How are we triggering this? Back then I think the idea was:

> As I said before, I want to use SQS to push things into levriero, not an API. Pull for backfilling and some agents, push via SQS if possible

So I did something similar to the other agents are doing. As far as I can see crossref agent goes to XrefEventData, the related_identifier goes to SORL. So the usageUpdate agent goes to SQS (queue: stage_usage). That's where SaSHIMI is pushing every new report and new update.

**Pull request merged**